webapi/src/test/scala/org/knora/webapi/responders/v2/ValuesResponderV2Spec.scala

```diff
@@ -1742,18 +1746,168 @@ class ValuesResponderV2Spec extends CoreSpec() with ImplicitSender {
+
+            val valueFromTriplestore = getValue(
+                resourceIri = resourceIri,
+                maybePreviousLastModDate = maybeResourceLastModDate,
+                propertyIriForGravsearch = propertyIri,
+                propertyIriInResult = propertyIri,
+                expectedValueIri = stillImageFileValueIri.get,
+                requestingUser = anythingUser1
+            )
+
+            valueFromTriplestore.valueContent match {
+                case savedValue: StillImageFileValueContentV2 => savedValue should ===(valueContent)
+                case _ => throw AssertionException(s"Expected file value, got $valueFromTriplestore")
+            }
+        }
+
+        "not create a duplicate still image file value" in {
+            val resourceIri: IRI = aThingPictureIri
+            val propertyIri: SmartIri = OntologyConstants.KnoraApiV2WithValueObjects.HasStillImageFileValue.toSmartIri
+
+            val valueContent = StillImageFileValueContentV2(
+                ontologySchema = ApiV2WithValueObjects,
+                fileValue = FileValueV2(
+                    internalFilename = "IQUO3t1AABm-FSLC0vNvVpr.jp2",
+                    internalMimeType = "image/jp2",
+                    originalFilename = "test.tiff",
+                    originalMimeType = "image/tiff"
+                ),
+                dimX = 512,
+                dimY = 256
+            )
+
+            actorUnderTest ! CreateValueRequestV2(
+                CreateValueV2(
+                    resourceIri = resourceIri,
+                    resourceClassIri = "http://0.0.0.0:3333/ontology/0001/anything/v2#ThingPicture".toSmartIri,
+                    propertyIri = propertyIri,
+                    valueContent = valueContent
+                ),
+                requestingUser = anythingUser1,
+                apiRequestID = UUID.randomUUID
+            )
+
+            expectMsgPF(timeout) {
+                case msg: akka.actor.Status.Failure =>
+                    msg.cause.isInstanceOf[DuplicateValueException] should ===(true)
+            }
+        }
+
+        "return a Sipi error if Sipi fails to move a file to permanent storage" in {
+            val resourceIri: IRI = aThingPictureIri
+            val propertyIri: SmartIri = OntologyConstants.KnoraApiV2WithValueObjects.HasStillImageFileValue.toSmartIri
+
+            val valueContent = StillImageFileValueContentV2(
+                ontologySchema = ApiV2WithValueObjects,
+                fileValue = FileValueV2(
+                    internalFilename = MockSipiResponderV2.FAILURE_FILENAME, // tells the mock Sipi responder to simulate failure
+                    internalMimeType = "image/jp2",
+                    originalFilename = "test.tiff",
+                    originalMimeType = "image/tiff"
+                ),
+                dimX = 512,
+                dimY = 256
+            )
+
+            // Knora will accept this request.
```

Comment:
...but the mock sipi responder will fail.

Reply:
https://github.com/dhlab-basel/Knora/pull/1011/commits/f588bd05e55e6b4b3efa4eb91eaed718acd65c22